### ISSUE TITLE
Saving a DataFrame with many columns throws obscure error

### ISSUE THREAD

**Issue by skleinbo** [JuliaData/CSV.jl#635]
__Issue:__
The following code
``` julia
using DataFrames, CSV

df = DataFrame(rand(6, 10^5));
CSV.write("huge.csv", df);
```
produces

```
ERROR: LoadError: UndefVarError: x2465 not defined
Stacktrace:
 [1] x2466 at .\x2469:95 [inlined]
 [2] macro expansion at .\x2468:83 [inlined]
 [3] eachcolumn at .\x2468:66 [inlined]
 [4] writerow(`...
```
__Info:__
```julia
julia> versioninfo()
Julia Version 1.4.2
Commit 44fa15b150* (2020-05-23 18:35 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-8.0.1 (ORCJIT, skylake)

(@v1.4) pkg> st --manifest CSV DataFrames
Status `C:\Users\stephan\.julia\environments\v1.4\Manifest.toml`
  [336ed68f] CSV v0.6.2 [`C:\Users\stephan\.julia\dev\CSV`]
  [324d7699] CategoricalArrays v0.8.1
  [34da2185] Compat v3.10.0
  [9a962f9c] DataAPI v1.3.0
  [a93c6f00] DataFrames v0.21.2
  [48062228] FilePathsBase v0.8.0
  [41ab1584] InvertedIndices v1.0.0
  [82899510] IteratorInterfaceExtensions v1.0.0
  [e1d29d7a] Missings v0.4.3
  [69de0a69] Parsers v1.0.4
  [2dfb63ee] PooledArrays v0.5.3
  [189a3867] Reexport v0.2.0
  [a2af1166] SortingAlgorithms v0.3.1
  [3783bdb8] TableTraits v1.0.0
  [bd369af6] Tables v1.0.4
  [ea10d353] WeakRefStrings v0.6.2
  [ade2ca70] Dates
  [9fa8497b] Future
  [a63ad114] Mmap
  [de0858da] Printf
  [3fa0cd96] REPL
  [10745b16] Statistics
  [4ec0a83e] Unicode
```
Full error message - which is very long due to printing the many column names - can be found [here](https://gist.github.com/skleinbo/a1250fb6c64232f572b25b0887909f67)

- Works with 5*10^4 columns (buffer overflow?)
- Reproducible on Mac with same Julia/package versions
- Reproducible on Julia 1.3.1
- No changes were made to the CSV.jl source, despite being checked out for development.

**Comment by quinnj** [comment]
I've tried to dig into this a couple of times, but unfortunately, this is a pretty deep, nasty corrupted compiler issue. I've chatted with @keno a bit about what's going on and he said he'll try to take a look soonish. If we can narrow down a bit what exactly is causing corruption, maybe we can find a work-around, or perhaps there are concrete fixes in the compiler that can help.

FWIW, I can reproduce the results on latest 1.6 master with just hte following:
```julia
using CSV, DataFrames
ncols = 67000
df = DataFrame(rand(6, ncols));
CSV.write("huge.csv", df);
```
